crowbar_framework/app/controllers/api/restart_management_controller.rb

```diff
@@ -81,19 +82,35 @@ def restarts_get
 
   def restarts_post
     params.require(:node)
-    params.require(:service)
 
     node_name = params[:node]
+    cookbook = params[:cookbook]
     service = params[:service]
 
     node = get_node_or_raise(node_name)
 
-    managed_cookbooks.each do |cookbook|
-      next unless node.key? :crowbar_wall
-      next unless node[:crowbar_wall].key? :requires_restart
-      next unless node[:crowbar_wall][:requires_restart].key? cookbook
-      next unless node[:crowbar_wall][:requires_restart][cookbook].key? service
-      node[:crowbar_wall][:requires_restart][cookbook].delete(service)
+    raise Crowbar::Error::NotFound unless
+      node.key?(:crowbar_wall) && node[:crowbar_wall].key?(:requires_restart)
```

Comment:
Why do we raise an error here? I would just pretend it all went fine.

Reply:
I thought that if users make a mistake, if it doesn't return an error they could think that the operation has executed correctly.

However, WIP